src/ngraph/pass/opset1_downgrade.cpp

```diff
@@ -0,0 +1,98 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/pass/opset1_downgrade.hpp"
+#include "ngraph/graph_util.hpp"
+#include "ngraph/op/get_output_element.hpp"
+#include "ngraph/op/pad.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+#define NGRAPH_OP(a, b) a,
+enum class OP_TYPEID
+{
+#include "ngraph/op/op_tbl.hpp"
+};
+#undef NGRAPH_OP
+
+#define NGRAPH_OP(a, b) {#a, OP_TYPEID::a},
+static unordered_map<string, OP_TYPEID> typeid_map{
+#include "ngraph/op/op_tbl.hpp"
+};
+#undef NGRAPH_OP
+
+static OP_TYPEID get_typeid(shared_ptr<Node> node)
+{
+    OP_TYPEID type_id;
+    auto it = typeid_map.find(node->description());
+    if (it != typeid_map.end())
+    {
+        type_id = it->second;
+    }
+    else
+    {
+        throw unsupported_op("Unsupported op '" + node->description() + "'");
+    }
+    return type_id;
+}
+// END mapping to OP_TYPEID
+
+bool pass::Opset1Downgrade::run_on_node(shared_ptr<Node> node)
+{
+    bool modified = false;
+
+    size_t op_version = node->get_version();
+
+    if (op_version == 0)
+    {
+        return modified;
+    }
+
+    NGRAPH_CHECK(op_version == 1,
+                 "Op version 1 transformation pass failed for ",
+                 *node,
+                 ", only op version 1 operations expected. Op version ",
+                 op_version,
+                 " found.");
+
+// Not all enumeration values explicitly handled in switch
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wswitch-enum"
+#endif
+    switch (get_typeid(node))
+    {
+    case OP_TYPEID::Pad:
+    {
+        auto tmp = dynamic_cast<const op::v1::Pad*>(node.get());
+        auto replacement_node = make_shared<op::v0::Pad>(node->input(0).get_source_output(),
```

Comment:
Let's assign `node->input(0).get_source_output()` and `node->input(3).get_source_output()` to variables to make code self-documenting.